internal/clients/gitdiff/gitdiff.go

```diff
@@ -0,0 +1,88 @@
+package gitdiff
+
+import (
+	"bytes"
+
+	"github.com/nightfallai/jenkins_test/internal/clients/diffreviewer"
+	"github.com/nightfallai/jenkins_test/internal/interfaces/libgitintf"
+)
+
+// Client is a wrapper around the libgit interface
+type Client struct {
+	Libgit       libgitintf.Libgit
+	RepoFilePath string
+}
+
+// DiffOptions options specifying the file diffs to be returned
+type DiffOptions struct {
+	FilterLineType map[diffreviewer.LineType]bool
+}
+
+// NewClient creates a libgit client
+func NewClient(accessToken, repoFilePath string) *Client {
+	return &Client{
+		Libgit:       NewLibgit(accessToken),
+		RepoFilePath: repoFilePath,
+	}
+}
+
+// GetDiff gets the diff from the base to the head on the given repo
+func (c *Client) GetDiff(baseRev, headRev, repoURL string, diffOpts *DiffOptions) ([]*diffreviewer.FileDiff, error) {
+	repo, err := c.Libgit.Clone(repoURL, c.RepoFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	diff, err := c.Libgit.DiffRevToRev(repo, baseRev, headRev)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
what's the difference between a diff and a patch?

Reply:
The patch is just the format of string that we want to get the diff in. Patch gives us the reguar diff that we get with git diff. The other option I was looking at was Raw but that just gives like a summary of the files changed.

Patch
```
diff --git a/clients/cassandra.go b/clients/cassandra.go
index 94cbad2..ae43ee5 100644
--- a/clients/cassandra.go
+++ b/clients/cassandra.go
@@ -44,8 +44,8 @@ func NewCassandra(
 		cluster.DisableInitialHostLookup = true
 	}

```
Raw
```
:100644 100644 94cbad2... ae43ee5... M	clients/cassandra.go
:100644 100644 05b045a... 3ae0772... M	clients/kafka.go
:100644 100644 9a55398... 182ad8c... M	config/config.go
:100644 100644 5db55b5... 739afc5... M	config/config_test.go
```